src/torrent.cpp

```diff
@@ -564,6 +564,15 @@ bool is_downloading_state(int const st)
 					? m_peer_list->num_connect_candidates() : -1);
 			}
 #endif
+            
+            if (!p.ca_certificate.empty()) {
+                m_ssl_torrent = true;
+    #ifdef TORRENT_SSL_PEERS
+                init_ssl(p.ca_certificate);
```

Comment:
I think it would be better to use the same path as when the cert is part of the torrent file itself. This looks shady.

For example, what happens if you load an SSL torrent (with an embedded CA cert) and also set this `add_torrent_params` field?

Reply:
that is a tricky one. I'll see what I can do.
BTW thanks a lot for the feedback.